### DISCUSSION TITLE
[Feature] Release helm chart to https://artifacthub.io

### DISCUSSION THREAD

**Discussion by joelee2012** [apache/skywalking#11693] [General]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

so far the helm chart is released to oci://registry-1.docker.io/apache/skywalking-helm, which is not accessed in china default and not suitable for helm chart. [artifacthub](https://artifacthub.io) is a dedicated hub for Kubernetes packages which provide more functionality for helm chart.

### Related issues

```
helm repo add skywalking-oci oci://registry-1.docker.io/apache/skywalking-helm
Error: looks like "oci://registry-1.docker.io/apache/skywalking-helm" is not a valid chart repository or cannot be reached: object required
```

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
There could be many vendors back on the helm or image services. In China, typically, Aliyun could do that as well.
I can not see why this vendor should be involved in the ASF release process. Docker is more common, and if you need a China mirror, it could be anyone.

**Reply by joelee2012** [reply]
Artifacthub belong to CNCF and provides the whole ecosystem for kubernetes packages.  Use mirror will involve additional complex for user

**Reply by wu-sheng** [reply]
`Artifact Hub is a CNCF sandbox project.` The project means source codes, rather than service itself. 

I can the copyright is Linux Foundation. This is good though. A foundation is better than a lock-in verndor.

**Comment by Superskyyy** [comment]
I would say I support releasing charts to artifacthub, just my opinion that it benefits from auto cve scans via Trivy and being a defacto registry for helm charts nowadays. we can simply set it up to process from the existing oci docker repo and no other maintenance overheads. (https://artifacthub.io/docs/topics/repositories/helm-charts/#oci-support)

**Reply by wu-sheng** [reply]
I think the first thing we need to get involved in is, the account creation and management of that new place. 
Maintenance overhead is not only about the manual steps in the release process.

So, if you or someone else wants to take ownership of this, we at least should have answers of
1. Why do you need this? Why is not suitable? `In China`, does it only mean something about the network?
1. Be clear about what options do we have? We post docker images, so we picked docker hub, which was/is a straightforward option, if you offer more, you should bring up the reasons, options, cons and pros of those options.

I think this is not about being supported or not. The initiative motivation is not clear after all.

**Comment by wu-sheng** [comment]
@joelee2012 @Superskyyy Please comment here and list what are you proposing to change. We need to keep the whole PMC in the loop and make this decision open.

**Comment by kezhenxu94** [comment]
Using oci-based registries is a common issue in Helm, not just Docker-specific, you actually don't need to run `helm repo add ...` first, if you want to install, just issue command `helm install oci://...` as documented in our Helm repo, if you want to add in the `Chart.yaml` as a dependency, just put the `oci://...` link in the dependency, read https://helm.sh/docs/topics/registries/ for more detail.

To me, "which is not accessed in china default", this is not a reason why we should do X, GitHub is not accessible in China, and you managed to open an issue here, maven central is not accessible in China, and people can find other vendor mirrors, we as a community are not going to solve network issues for the users. And finally, you just CAN'T guarantee that the GFW won't block https://artifacthub.io/

**Reply by kezhenxu94** [reply]
Note that I'm not against to deploy Helm Chart to https://artifacthub.io/, instead, I'm wondering what it brings to the users that Docker Hub or GitHub Registry cannot.

**Comment by joelee2012** [comment]
Here is difference between them: 
- https://artifacthub.io/docs/topics/repositories/helm-charts/#helm-charts-repositories 
- https://helm.sh/blog/storing-charts-in-oci/
or you just try Artifacthub to find difference and experience by yourself
The network issue is not only reason, i just propose a choice to make thing better rather than convincing someone

**Reply by wu-sheng** [reply]
> propose a choice to make thing better rather than convincing someone

You are convincing the team, and especially, you need to convince the release manager(s) of that repository.
Because in the ASF, release would never be automatically by a bot. It is always by a human being. So, the release process will only be changed until the major release managers of that repo(https://github.com/apache/skywalking-helm/releases) agree.
In SkyWalking, those who take action are the people making the call. 

> The network issue is not only reason

Your proposal was very clear this is the reason, especially Zhenxu has shared how dockerhub helm oci works.

**Comment by freshgeek** [comment]
I don't care about how to provide it. May I ask if there is a solution to the issue of installing Oci Helm in the China region？

**Reply by freshgeek** [reply]
https://github.com/helm/helm/issues/10065

**Reply by freshgeek** [reply]
https://helm.sh/docs/topics/registries/
